### PR TITLE
feat: configure git for floxmeta

### PR DESCRIPTION
**Pull request by ysndr**
* Configure git author data
* Redirect git global and system config files to /dev/null
* Pass floxhub token through `Flox` to floxmeta constructors and configure credential helper

### CODE REVIEW

File: crates/flox-rust-sdk/src/models/floxmetav2/mod.rs [hunk 1]
```diff
@@ -67,6 +81,31 @@ impl FloxmetaV2 {
...
+    // unset the global and system config
+    options.add_env_var("GIT_CONFIG_GLOBAL", "/dev/null");
+    options.add_env_var("GIT_CONFIG_SYSTEM", "/dev/null");
+
+    // Set authentication with the floxhub token using an inline credential helper.
+    // The credential helper should help avoinding a leak of the token in the process list.
+    options.add_env_var("FLOX_FLOXHUB_TOKEN", floxhub_token);
+    options.add_config_flag(
+        &format!("credentials.{floxhub_host}.helper"),
+        "!f(){ echo ${FLOX_FLOXHUB_TOKEN}; }; f",
```
Comment (ysndr) [inline crates/flox-rust-sdk/src/models/floxmetav2/mod.rs:103]: If we don't want to bubble down the token we could make this 

```suggestion
        "!f(){ $out/bin/flox auth token }; f",
```

but I think the error messages git would give us are worse than managing the token ourselves.

File: crates/flox-rust-sdk/src/providers/git.rs [hunk 2]
```diff
@@ -140,66 +204,85 @@ impl GitCommandProvider {
     }
 
     /// Open a repo, erroring if `path` is not a repo or is a subdirectory of a repo
-    //
-    // TODO should share more code with discover?
-    pub fn open<P: AsRef<Path>>(path: P) -> Result<Self, GitCommandOpenError> {
-        let out_str = GitCommandProvider::run_command(
-            GitCommandProvider::new_command(&Some(&path))
+    pub fn open_with<P: AsRef<Path>>(
```
Comment (zmitchell) [inline crates/flox-rust-sdk/src/providers/git.rs:207]: nitpick: I think `open_with_options` is a slightly clearer name

**Review by zmitchell — APPROVED** [review]

File: crates/flox-rust-sdk/src/models/floxmetav2/mod.rs (hunk 1 above)
Comment (zmitchell) [reply crates/flox-rust-sdk/src/models/floxmetav2/mod.rs:103]: Current implementation looks fine to me!

File: crates/flox-rust-sdk/src/providers/git.rs (hunk 2 above)
Comment (zmitchell) [reply crates/flox-rust-sdk/src/providers/git.rs:207]: Ping me if you don't plan on renaming and I'll merge.

Comment (ysndr) [reply crates/flox-rust-sdk/src/providers/git.rs:207]: I dont plan for it now.
If its itching or get ambiguous in some commands i'll consider making it more explicit

**Pull request merged**